viper_test.go

```diff
@@ -1349,6 +1349,86 @@ func TestWriteConfigYAML(t *testing.T) {
 	assert.Equal(t, yamlWriteExpected, read)
 }
 
+func TestSafeWriteConfig(t *testing.T) {
+	v := New()
+	fs := afero.NewMemMapFs()
+	v.SetFs(fs)
+	v.AddConfigPath("/test")
+	v.SetConfigName("c")
+	v.SetConfigType("yaml")
+	err := v.ReadConfig(bytes.NewBuffer(yamlExample))
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err = v.SafeWriteConfig(); err != nil {
+		t.Fatal(err)
+	}
+	read, err := afero.ReadFile(fs, "/test/c.yaml")
+	if err != nil {
+		t.Fatal(err)
+	}
+	assert.Equal(t, yamlWriteExpected, read)
+}
+
+func TestSafeWriteConfigWithMissingConfigPath(t *testing.T) {
+	v := New()
+	fs := afero.NewMemMapFs()
+	v.SetFs(fs)
+	v.SetConfigName("c")
+	v.SetConfigType("yaml")
+	err := v.SafeWriteConfig()
+	if err == nil {
+		t.Fatal("Expected exception")
```

Comment:
This is  not Java 🙂 There are no exceptions, but errors.

Reply:
> This is not Java 🙂 There are no exceptions, but errors.

LOL...you caught me.